tests/ops/op_math/test_sprod.py

```diff
@@ -0,0 +1,635 @@
+
+        sprod_op = SProd(42, U_op)
+        mat = sprod_op.matrix()
+
+        true_mat = 42 * U
+        assert np.allclose(mat, true_mat)
+
+    # TODO[Jay]: remove xfail once there is support for sparse matrices for most operations
+    @pytest.mark.xfail
+    @pytest.mark.parametrize("scalar, op", ops)
+    def test_sparse_matrix(self, op, scalar):
+        """Test the sparse_matrix representation of scaled ops."""
+        sprod_op = SProd(scalar, op)
+        sparse_matrix = sprod_op.sparse_matrix()
+
+        expected_sparse_matrix = scalar * op.matrix()
+        expected_sparse_matrix = csr_matrix(expected_sparse_matrix)
+
+        assert np.allclose(sparse_matrix, expected_sparse_matrix)
+
+    # Add interface tests for each interface !
+
+    @pytest.mark.jax
+    def test_sprod_jax(self):
+        """Test matrix is cast correctly using jax parameters."""
+        import jax.numpy as jnp
+
+        coeff = jnp.array(1.23)
+        rot_params = jnp.array([0.12, 3.45, 6.78])
+
+        sprod_op = SProd(coeff, qml.Rot(rot_params[0], rot_params[1], rot_params[2], wires=0))
+        mat = sprod_op.matrix()
+
+        true_mat = 1.23 * gd.Rot3(rot_params[0], rot_params[1], rot_params[2])
+        true_mat = jnp.array(true_mat)
+
+        assert jnp.allclose(mat, true_mat)
+
+    @pytest.mark.torch
+    def test_sprod_torch(self):
+        """Test matrix is cast correctly using torch parameters."""
+        import torch
+
+        coeff = torch.tensor(1.23)
+        rot_params = torch.tensor([0.12, 3.45, 6.78])
+
+        sprod_op = SProd(coeff, qml.Rot(rot_params[0], rot_params[1], rot_params[2], wires=0))
+        mat = sprod_op.matrix()
+
+        true_mat = 1.23 * gd.Rot3(rot_params[0], rot_params[1], rot_params[2])
+        true_mat = torch.tensor(true_mat, dtype=torch.complex64)
+
+        assert torch.allclose(mat, true_mat)
+
+    @pytest.mark.tf
+    def test_sprod_tf(self):
+        """Test matrix is cast correctly using tf parameters."""
+        import tensorflow as tf
+
+        coeff = tf.Variable(1.23, dtype=tf.complex128)
+        raw_rot_params = [0.12, 3.45, 6.78]
+        rot_params = tf.Variable(raw_rot_params)
+
+        sprod_op = SProd(coeff, qml.Rot(rot_params[0], rot_params[1], rot_params[2], wires=0))
+        mat = sprod_op.matrix()
+
+        true_mat = 1.23 * gd.Rot3(raw_rot_params[0], raw_rot_params[1], raw_rot_params[2])
+        true_mat = tf.Variable(true_mat, dtype=tf.complex128)
+
+        assert isinstance(mat, tf.Tensor)
+        assert mat.dtype == true_mat.dtype
+        assert np.allclose(mat, true_mat)
+
+
+class TestProperties:
+    @pytest.mark.parametrize("op_scalar_tup", ops)
+    def test_queue_catagory(self, op_scalar_tup):
+        """Test queue_catagory property is always None."""  # currently not supporting queuing SProd
+        scalar, op = op_scalar_tup
+        sprod_op = SProd(scalar, op)
+        assert sprod_op._queue_category is None
+
+    def test_diagonalizing_gates(self):
```

Comment:
This is under the test section `TestProperties`, but `diagonalizing_gates` is a method.

Reply:
Moved to `Initialization` heading